File: src/Hedwig/Models/Child.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Hedwig.Validations;
using Hedwig.Validations.Attributes;
using Hedwig.Repositories;
using Microsoft.AspNetCore.Http;

namespace Hedwig.Models
{
	public class Child : TemporalEntity, IHedwigIdEntity<Guid>, INonBlockingValidatableObject
	{
		[Required]
		public Guid Id { get; set; }

		public string Sasid { get; set; }

		[Required]
		[StringLength(35)]
		public string FirstName { get; set; }

		[StringLength(35)]
		public string MiddleName { get; set; }

		[Required]
		[StringLength(35)]
		public string LastName { get; set; }

		[StringLength(10)]
		public string Suffix { get; set; }
		public DateTime? Birthdate { get; set; }
		public string BirthTown { get; set; }
		public string BirthState { get; set; }
		public string BirthCertificateId { get; set; }
		public bool AmericanIndianOrAlaskaNative { get; set; } = false;
		public bool Asian { get; set; } = false;
		public bool BlackOrAfricanAmerican { get; set; } = false;
		public bool NativeHawaiianOrPacificIslander { get; set; } = false;
		public bool White { get; set; } = false;
		public bool? HispanicOrLatinxEthnicity { get; set; }

		[JsonConverter(typeof(StringEnumConverter))]
		public Gender Gender { get; set; } = Gender.Unspecified;
		public bool Foster { get; set; } = false;
		public int? FamilyId { get; set; }
		public Family Family { get; set; }
		public ICollection<Enrollment> Enrollments { get; set; }

		[Required]
		[OrgIdFromPath]
		public int OrganizationId { get; set; }
		public Organization Organization { get; set; }

		[NotMapped]	
		public List<ValidationError> ValidationErrors { get; set; }
	}
}

File: src/Hedwig/ClientApp/src/tests/data/child.ts
import { Child, Gender } from '../../generated';
import emptyGuid from '../../utils/emptyGuid';

export const mockChild: Child = {
	id: emptyGuid(),
	firstName: 'Lily',
	middleName: 'Luna',
	lastName: 'Potter',
	birthdate: new Date('2016-12-12'),
	birthTown: 'Hogsmeade',
	birthState: 'CT',
	birthCertificateId: '123',
	nativeHawaiianOrPacificIslander: true,
	hispanicOrLatinxEthnicity: true,
	gender: Gender.Female,
	foster: false,
	familyId: 1,
	organizationId: 1,
	family: {
		id: 1,
		addressLine1: '4 Privet Drive',
		town: 'Hogsmeade',
		state: 'CT',
		zip: '77777',
		homelessness: false,
		organizationId: 1,
		determinations: [
			{
				id: 1,
				familyId: 1,
				income: 20000,
				numberOfPeople: 4,
				determinationDate: new Date('2019-01-01'),
			},
			{
				id: 2,
				familyId: 1,
				income: 22000,
				numberOfPeople: 5,
				determinationDate: new Date('2020-01-01'),
			},
		],
	},
	c4KCertificates: [],
};
